### PR TITLE
Spacing tweaks and improved trailing slash mixin

### PR DESCRIPTION
**Pull request by zeftilldeath**
## What does this change?
On the new header sub-menu it looks aesthetically better if words don't appear directly beneath the veggie burger. Currently to achieve that there is a right-margin: 

screen shot 2017-03-14 at 16 54 32

The problem with that, is when there are more links like 'clubs' in this example they wrap, despite there being room on the second row for them to be show.

To sort this out I've changed all links with trailing slashes to be inline-block with the font-size: 0; fix which allows for correct alignment when they wrap. They are no longer floats.

To further standardise all elements with trailing slashes I have made the menu unordered lists just like page tags and improved the logic of the trailing slash mixin to incorporate that.

## Does this affect other platforms - Amp, Apps, etc?
Yep, updated amp to use this logic also.

## Screenshots
This illustrates the tap targets and the pink line illustrates the spacer than knocks the links onto a new line.

screen shot 2017-03-15 at 14 07 25

screen shot 2017-03-15 at 14 28 56

### CODE REVIEW

File: common/app/views/fragments/amp/header.scala.html [hunk 1]
```diff
@@ -16,13 +16,17 @@ <h1 class="u-h">The Guardian</h1>
         </a>
 
         <nav class="header__nav">
+            <ul class="header__nav__primary-links">
             @NewNavigation.PrimaryLinks.map { link =>
```
Comment (NataliaLKB) [inline common/app/views/fragments/amp/header.scala.html]: Nit: the spacing is off here. make sure the opening and closing `ul` tags match!

File: common/app/views/fragments/nav/subSectionNav.scala.html [hunk 2]
```diff
@@ -3,18 +3,19 @@
...
-            <a class="@RenderClasses(Map(
-                    "tertiary-navigation__link" -> true,
-                    "tertiary-navigation__link--current-section" -> (id == link.uniqueSection)
-                ))"
-                href="@LinkTo(link.url)" data-link-name="nav2 : tertiary : @{ if(link.longTitle.isEmpty) link.title else link.longTitle}">
-                    @link.title
-            </a>
+            <li class="tertiary-navigation__link-item">
+                <a class="@RenderClasses(Map(
+                        "tertiary-navigation__link" -> true,
```
Comment (NataliaLKB) [inline common/app/views/fragments/nav/subSectionNav.scala.html]: separate to this PR maybe, but we really should change `tertiary` to `subnav` to keep names consistent. My fault!

File: common/app/views/fragments/amp/header.scala.html [hunk 3]
```diff
@@ -16,13 +16,17 @@ <h1 class="u-h">The Guardian</h1>
...
-                <a class="header__nav__link"
-                    href="@LinkTo(link.url)"
-                    data-link-name="amp : nav : @link.title">
-                        @link.title
-                </a>
+                <li class="header__nav__link-item">
```
Comment (gustavpursche) [inline common/app/views/fragments/amp/header.scala.html]: In BEM there is only one level of nesting allowed :) So this should better be `header__nav-link-item`. Applies to a few cases following aswell.

**Review by gu-stav — COMMENTED** [review]
Just a little BEM nit - rest looks great. 🌷

File: common/app/views/fragments/amp/header.scala.html (hunk 1 above)
Comment (zeftilldeath) [inline common/app/views/fragments/amp/header.scala.html]: Thanks

File: common/app/views/fragments/amp/header.scala.html (hunk 3 above)
Comment (zeftilldeath) [inline common/app/views/fragments/amp/header.scala.html]: Cheers, have changed them, please check to see I understood correctly.

**Review by NataliaLKB — APPROVED** [review]
lgtm 👍 !!

**Pull request merged**